lib/install/management/create.go

```diff
@@ -211,3 +213,69 @@ func (d *Dispatcher) uploadImages(files map[string]string) error {
 	}
 	return nil
 }
+
+// cleanupAfterCreationFailed cleans up the dangling resource pool for the failed VCH and any bridge network if there is any.
+// The function will not abort and early terminate upon any error during cleanup process. Error details are logged.
+func (d *Dispatcher) cleanupAfterCreationFailed(conf *config.VirtualContainerHostConfigSpec, cleanupNetwork bool) {
+	defer trace.End(trace.Begin("", d.op))
+	var err error
+
+	// clean up dangling empty resource pool
+	d.op.Info("Clean up dangling VCH resource pool after VCH creation failed...")
```

Comment:
Is this going to be confusing for users? It seems like we may be cleaning up the dangling resource pool between when the failure occurs and when we tell them that the failure occurred.

Reply:
Thanks, good point. Most of the times in the creation steps, the error message is logged before the creation exists and the message `Clean up dangling VCH resource...` will get logged after the failed error message, so the user knows what's going on. But we can't guarantee that it's always the case.
I'll add logging before we cleanup the dangling resource telling user what's failed.